src/preferences/dialog/dlgprefsound.cpp

```diff
@@ -312,9 +312,11 @@ void DlgPrefSound::slotApply() {
     SoundDeviceStatus status = SoundDeviceStatus::Ok;
     {
         ScopedWaitCursor cursor;
-        m_pKeylockEngine->set(keylockComboBox->currentData().toDouble());
+        const auto keylockEngine =
+                keylockComboBox->currentData().value<EngineBuffer::KeylockEngine>();
+        m_pKeylockEngine->set(static_cast<double>(keylockEngine));
```

Comment:
interesting, apparently `QVariant::toDouble` doesn't work if the contained type is an enum? https://doc.qt.io/qt-5/qvariant.html#toDouble 

Am I reading that correctly?